--- /home/jd12g09/install/tinker713_original/tinker/source/cutoffs.f	2015-05-08 21:57:53.000000000 +0100
+++ /home/jd12g09/install/tinker70/source/patched/cutoffs.f	2016-01-14 14:51:06.000000000 +0000
@@ -36,2 +36,2 @@
-      character*120 record
-      character*120 string
+      character*12000 record
+      character*12000 string
@@ -89 +89 @@
-         string = record(next:120)
+         string = record(next:12000)
