--- /home/jd12g09/install/tinker713_original/tinker/source/ehal3.f	2015-05-08 21:57:54.000000000 +0100
+++ /home/jd12g09/install/tinker70/source/patched/ehal3.f	2017-04-20 14:40:46.000000000 +0100
@@ -98,0 +99 @@
+      real*8 erep,edisp
@@ -232,0 +234,4 @@
+                     erep = eps * rv7 * tau**7
+     &                      * ((ghal+1.0d0)*rv7/rho)
+                     edisp = eps * rv7 * tau**7
+     &                      * (-2.0d0)
@@ -281 +286 @@
-     &                                rv,sqrt(rik2),e
+     &                                rv,sqrt(rik2),e,erep,edisp
@@ -283 +288 @@
-     &                          13x,2f10.4,f12.4)
+     &                          13x,2f10.4,f17.9,f17.9,f17.9)
@@ -445 +450 @@
-     &                             '(XTAL)',4x,2f10.4,f12.4)
+     &                             '(XTAL)',4x,2f10.4,f17.9)
@@ -768 +773 @@
-     &                             13x,2f10.4,f12.4)
+     &                             13x,2f10.4,f17.9)
@@ -773 +778 @@
-     &                             '(XTAL)',4x,2f10.4,f12.4)
+     &                             '(XTAL)',4x,2f10.4,f17.9)
@@ -1063 +1068 @@
-     &                          13x,2f10.4,f12.4)
+     &                          13x,2f10.4,f17.9)
