--- /home/jd12g09/install/tinker713_original/tinker/source/getkey.f	2015-05-08 21:57:54.000000000 +0100
+++ /home/jd12g09/install/tinker70/source/patched/getkey.f	2016-01-14 15:47:00.000000000 +0000
@@ -32,4 +32,4 @@
-      character*120 keyfile
-      character*120 comment
-      character*120 record
-      character*120 string
+      character*12000 keyfile
+      character*12000 comment
+      character*12000 record
+      character*12000 string
@@ -86 +86 @@
-   20       format (a120)
+   20       format (a12000)
@@ -109 +109 @@
-            comment = record(next:120)
+            comment = record(next:12000)
@@ -133 +133 @@
-         string = record(next:120)
+         string = record(next:12000)
