--- /home/jd12g09/install/tinker713_original/tinker/source/poledit.f	2015-05-08 21:57:53.000000000 +0100
+++ /home/jd12g09/install/tinker70/source/patched/poledit.f	2016-11-29 09:36:20.000000000 +0000
@@ -185,0 +186 @@
+      onetep_gdma = .false.
@@ -188,0 +190,3 @@
+         if(record(36:41) .eq. 'ONETEP') then
+            onetep_gdma = .true.
+         end if
@@ -324 +328 @@
-  150    format (/,' Charge:',10x,f15.5)
+  150    format (/,' Charge:',10x,f17.7)
@@ -366,0 +371 @@
+
@@ -375 +380 @@
-         read (record(26:36),*)  mp(i)
+         read (record(26:45),*)  mp(i)
@@ -377 +382 @@
-         read (record(26:36),*)  dpz(i)
+         read (record(26:45),*)  dpz(i)
@@ -380 +385 @@
-         read (record(26:36),*)  dpx(i)
+         read (record(26:45),*)  dpx(i)
@@ -383 +388 @@
-         read (record(26:36),*)  dpy(i)
+         read (record(26:45),*)  dpy(i)
@@ -386 +391 @@
-         read (record(26:36),*)  q20(i)
+         read (record(26:45),*)  q20(i)
@@ -389 +394 @@
-         read (record(26:36),*)  q21c(i)
+         read (record(26:45),*)  q21c(i)
@@ -392 +397 @@
-         read (record(26:36),*)  q21s(i)
+         read (record(26:45),*)  q21s(i)
@@ -395 +400 @@
-         read (record(26:36),*)  q22c(i)
+         read (record(26:45),*)  q22c(i)
@@ -398 +403 @@
-         read (record(26:36),*)  q22s(i)
+         read (record(26:45),*)  q22s(i)
@@ -429,2 +434,2 @@
-      else if (record(39:42) .eq. 'Q11c') then
-         read (record(45:55),*)  dpx(i)
+      else if (record(48:51) .eq. 'Q11c') then
+         read (record(54:73),*)  dpx(i)
@@ -432,2 +437,2 @@
-      else if (record(39:42) .eq. 'Q11s') then
-         read (record(45:55),*)  dpy(i)
+      else if (record(48:51) .eq. 'Q11s') then
+         read (record(54:73),*)  dpy(i)
@@ -435,2 +440,2 @@
-      else if (record(39:42) .eq. 'Q21c') then
-         read (record(45:55),*)  q21c(i)
+      else if (record(48:51) .eq. 'Q21c') then
+         read (record(54:73),*)  q21c(i)
@@ -438,2 +443,2 @@
-      else if (record(39:42) .eq. 'Q21s') then
-         read (record(45:55),*)  q21s(i)
+      else if (record(48:51) .eq. 'Q21s') then
+         read (record(54:73),*)  q21s(i)
@@ -441,2 +446,2 @@
-      else if (record(39:42) .eq. 'Q22c') then
-         read (record(45:55),*)  q22c(i)
+      else if (record(48:51) .eq. 'Q22c') then
+         read (record(54:73),*)  q22c(i)
@@ -444,2 +449,2 @@
-      else if (record(39:42) .eq. 'Q22s') then
-         read (record(45:55),*)  q22s(i)
+      else if (record(48:51) .eq. 'Q22s') then
+         read (record(54:73),*)  q22s(i)
@@ -475,8 +480,8 @@
-      else if (record(58:61) .eq. 'Q11s') then
-         read (record(64:74),*)  dpy(i)
-      else if (record(58:61) .eq. 'Q21s') then
-         read (record(64:74),*)  q21s(i)
-      else if (record(58:61) .eq. 'Q22c') then
-         read (record(64:74),*)  q22c(i)
-      else if (record(58:61) .eq. 'Q22s') then
-         read (record(64:74),*)  q22s(i)
+      else if (record(76:79) .eq. 'Q11s') then
+         read (record(82:101),*)  dpy(i)
+      else if (record(76:79) .eq. 'Q21s') then
+         read (record(82:101),*)  q21s(i)
+      else if (record(76:79) .eq. 'Q22c') then
+         read (record(82:101),*)  q22c(i)
+      else if (record(76:79) .eq. 'Q22s') then
+         read (record(82:101),*)  q22s(i)
@@ -1159 +1164 @@
-   60       format (/,' Charge:',10x,f15.5)
+   60       format (/,' Charge:',10x,f17.7)
@@ -1337,7 +1342,7 @@
-c
-      eps = 0.00001d0
-      do i = 1, npole
-         do j = 1, 13
-            pole(j,i) = dble(nint(pole(j,i)/eps)) * eps
-         end do
-      end do
+c     jd: removed. Cannot increase beyond one further 0 due to int wraparound
+c      eps = 0.00001d0
+c      do i = 1, npole
+c         do j = 1, 13
+c            pole(j,i) = dble(nint(pole(j,i)/eps)) * eps
+c         end do
+c      end do
@@ -1401 +1406 @@
-  180       format (/,' Charge:',10x,f15.5)
+  180       format (/,' Charge:',10x,f17.7)
@@ -1622 +1627 @@
-      call interpol
+      if(.not. onetep_gdma) call interpol
@@ -1627,3 +1632,5 @@
-         rpole(2,i) = rpole(2,i) - uind(1,i)
-         rpole(3,i) = rpole(3,i) - uind(2,i)
-         rpole(4,i) = rpole(4,i) - uind(3,i)
+         if(.not. onetep_gdma) then
+            rpole(2,i) = rpole(2,i) - uind(1,i)
+            rpole(3,i) = rpole(3,i) - uind(2,i)
+            rpole(4,i) = rpole(4,i) - uind(3,i)
+         end if
@@ -1687 +1694 @@
-   60       format (/,' Charge:',10x,f15.5)
+   60       format (/,' Charge:',10x,f17.7)
@@ -2288,7 +2295,7 @@
-c
-      eps = 0.00001d0
-      do i = 1, npole
-         do j = 1, 13
-            pole(j,i) = dble(nint(pole(j,i)/eps)) * eps
-         end do
-      end do
+c     jd: removed. Cannot increase beyond one further 0 due to int wraparound
+c      eps = 0.0000000001d0
+c      do i = 1, npole
+c         do j = 1, 13
+c            pole(j,i) = dble(nint(pole(j,i)/eps)) * eps
+c         end do
+c      end do
@@ -2298,18 +2305,18 @@
-      k = 0
-      big = 0.0d0
-      sum = 0.0d0
-      do i = 1, npole
-         sum = sum + pole(1,i)
-         ci = abs(pole(1,i))
-         if (ci .gt. big) then
-            do j = 1, n
-               cj = abs(pole(1,j))
-               if (i.ne.j .and. ci.eq.cj)  goto 60
-            end do
-            k = i
-            big = ci
-   60       continue
-         end if
-      end do
-      sum = sum - dble(nint(sum))
-      if (k .ne. 0)  pole(1,k) = pole(1,k) - sum
+c      k = 0
+c      big = 0.0d0
+c      sum = 0.0d0
+c      do i = 1, npole
+c         sum = sum + pole(1,i)
+c         ci = abs(pole(1,i))
+c         if (ci .gt. big) then
+c            do j = 1, n
+c               cj = abs(pole(1,j))
+c               if (i.ne.j .and. ci.eq.cj)  goto 60
+c            end do
+c            k = i
+c            big = ci
+c   60       continue
+c         end if
+c      end do
+c      sum = sum - dble(nint(sum))
+c      if (k .ne. 0)  pole(1,k) = pole(1,k) - sum
@@ -2353 +2360 @@
-  120       format (/,' Charge:',10x,f15.5)
+  120       format (/,' Charge:',10x,f17.7)
@@ -2451 +2458 @@
-   50       format ('multipole',1x,i5,21x,f11.5)
+   50       format ('multipole',1x,i5,21x,f21.15)
@@ -2454 +2461 @@
-   60       format ('multipole',1x,2i5,16x,f11.5)
+   60       format ('multipole',1x,2i5,16x,f21.15)
@@ -2458 +2465 @@
-   70          format ('multipole',1x,3i5,11x,f11.5)
+   70          format ('multipole',1x,3i5,11x,f21.15)
@@ -2461 +2468 @@
-   80          format ('multipole',1x,4i5,6x,f11.5)
+   80          format ('multipole',1x,4i5,6x,f21.15)
@@ -2466 +2473 @@
-   90          format ('multipole',1x,3i5,11x,f11.5)
+   90          format ('multipole',1x,3i5,11x,f21.15)
@@ -2469 +2476 @@
-  100          format ('multipole',1x,4i5,6x,f11.5)
+  100          format ('multipole',1x,4i5,6x,f21.15)
@@ -2473 +2480 @@
-  110       format ('multipole',1x,4i5,6x,f11.5)
+  110       format ('multipole',1x,4i5,6x,f21.15)
@@ -2476 +2483 @@
-  120       format ('multipole',1x,4i5,6x,f11.5)
+  120       format ('multipole',1x,4i5,6x,f21.15)
@@ -2479 +2486 @@
-  130    format (36x,3f11.5)
+  130    format (36x,3f21.15)
@@ -2481 +2488 @@
-  140    format (36x,f11.5)
+  140    format (36x,f21.15)
@@ -2483 +2490 @@
-  150    format (36x,2f11.5)
+  150    format (36x,2f21.15)
@@ -2485 +2492 @@
-  160    format (36x,3f11.5)
+  160    format (36x,3f21.15)
